### PR TITLE
Convert suite settings (coins, custom-firmware) + refactors desktop pipeline

### PR DESCRIPTION
**Pull request by Vere-Grey**
Converts:
settings/coins.test.ts
settings/custom-firmware.test.ts

Fixes:
settings/t2b1-device-settings.test.ts

Refactors and updates desktop pipeline. Works with tags and is updated according to the web one,

### CODE REVIEW

File: .github/workflows/test-suite-web-e2e-pw.yml [hunk 1]
```diff
@@ -144,21 +136,18 @@ jobs:
...
-          TEST_GROUP: ${{ matrix.TEST_GROUP }}
           GITHUB_ACTION: true
           CURRENTS_PROJECT_ID: Og0NOQ
           CURRENTS_RECORD_KEY: ${{ secrets.CURRENTS_RECORD_KEY }}
           CURRENTS_CI_BUILD_ID: pr-run-${{github.run_id}}
         run: |
           docker compose up -d ${{ matrix.CONTAINERS }}
-          echo "Starting Playwright Web test group ${TEST_GROUP}"
-          yarn workspace @trezor/suite-desktop-core test:e2e:web --grep=${TEST_GROUP}
+          echo "Starting Playwright Web test group ${{ matrix.TEST_GROUP }}"
```
Comment (Vere-Grey) [inline .github/workflows/test-suite-web-e2e-pw.yml:149]: Unify how vars are passed to bash and adds safety net --forbid-only. Playwright docs: Whether to disallow test.only. Useful on CI.

File: .github/workflows/test-suite-desktop-e2e.yml [hunk 2]
```diff
@@ -27,78 +27,87 @@ jobs:
...
+          #   CONTAINERS: "trezor-user-env-unix"
+          # - TEST_GROUP: "@group=passphrase"
+          #   CONTAINERS: "trezor-user-env-unix"
+          # - TEST_GROUP: "@group=other"
+          #   CONTAINERS: "trezor-user-env-unix"
+          - TEST_GROUP: "@group=wallet"
+            CONTAINERS: "trezor-user-env-unix bitcoin-regtest"
 
     steps:
       - name: Checkout
```
Comment (Vere-Grey) [inline .github/workflows/test-suite-desktop-e2e.yml:46]: Our web pipeline uses different checkout:
```
      - name: Checkout
        uses: actions/checkout@v4
        with:
          ref: ${{github.event.after}}
          fetch-depth: 2
```

Any preferences?

File: packages/suite-desktop-core/e2e/playwright.config.ts [hunk 3]
```diff
@@ -34,7 +36,7 @@ const config: PlaywrightTestConfig = {
     reporter: process.env.GITHUB_ACTION
         ? [['list'], ['@currents/playwright'], ['html', { open: 'never' }]]
         : [['list'], ['html', { open: 'never' }]],
-    timeout: 1000 * 60 * 5,
+    timeout: process.env.GITHUB_ACTION ? timeoutCIRun : timeoutLocalRun,
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/playwright.config.ts:39]: We want locally tests to fail fast during troubleshooting.

File: packages/suite-desktop-core/e2e/support/common.ts [hunk 4]
```diff
@@ -100,3 +102,14 @@ export const launchSuite = async (params: LaunchSuiteParams = {}) => {
...
+export const getApiUrl = (webBaseUrl: string | undefined, testInfo: TestInfo) => {
+    const electronApiURL = 'file:///';
+    const apiURL =
+        testInfo.project.name === PlaywrightProjects.Desktop ? electronApiURL : webBaseUrl;
+    if (!apiURL) {
+        throw new Error('apiURL is not defined');
+    }
+
+    return apiURL;
+};
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/support/common.ts:115]: might need update once i see first ci run

File: packages/suite-desktop-core/e2e/tests/settings/custom-firmware.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            await page.getByTestId('@settings/device/custom-firmware-modal-button').click();
+            await expect(page.getByTestId('@firmware-modal/install-button')).toBeDisabled();
+        });
+
+        await test.step('Select the custom firmware', async () => {
+            const fileChooserPromise = page.waitForEvent('filechooser');
+            await page.getByTestId('@firmware-modal/input-area').click();
+            const fileChooser = await fileChooserPromise;
+            await fileChooser.setFiles(firmwarePath);
+        });
```
Comment (Vere-Grey) [inline packages/suite-desktop-core/e2e/tests/settings/custom-firmware.test.ts:26]: I am really impressed this works in desktop. Well done playwright

**Review by HajekOndrej — APPROVED** [review]

**Pull request merged**